README.md

```diff
@@ -425,3 +431,381 @@ Now that it's done, if we go back to the repository the python script will be vi
+
+- **For macOS users:**
+  - Built-in: Terminal. Press ⌘ command + space and type “terminal” to find it.
+  - iTerm2, which you can integrate with zsh and oh my zsh for color highlighting, among other handy features for Git users.
+- **For Windows users:**
+  - Built-in: cmd. Click the search icon on the bottom navigation bar on Windows and type cmd to find it.
+  - PowerShell: a Windows “powered up” shell, from which you can execute a greater number of commands.
+  - Git Bash: it comes built into Git for Windows.
+- **For Linux users:**
+  - Built-in: Linux Terminal.
+
+### Installing and configuring git on the terminal ### 
+
+There are a lot of resources online which will guide you how to download, insntall and configure git on your local machine. A quick Google search ‘How to install git on [your preferred OS]” will reveal a number of resources. Here is a useful link: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+
+#### Configure Git: ####
+
+To start using Git from your computer, you must enter your credentials (user name and email) to identify you as the author of your work.
+
+In your shell, add your user name:
+
+```bash
+$ git config --global user.name "your_username"
+```
+
+And your email address:
+
+```bash
+$ git config --global user.email "your_email_address@example.com"
+```
+
+To check the configuration, run:
+
+```bash
+$ git config --global --list
+```
+
+The --global option tells Git to always use this information for anything you do on your system. If you omit --global or use --local, the configuration is applied only to the current repository.
+
+
+### Setting up your local git repository ###
+
+There maybe a time when you do not have a remote repository to start off with or that you would like to create your own repository and share with your team.
+
+The first step is to create project directory and cd into it. Next, run the following command:
+
+```bash
+$ git init
+```
+The git init command creates a new Git repository. This command turns a directory into an empty Git repository. This is the first step in creating a repository. After running git init, adding and committing files/directories becomes possible.
+
+Usage:
+
+```bash
+# change directory to codebase
+$ cd /file/path/to/code
+# make directory a git repository
+$ git init
+Initialized empty Git repository in /file/path/to/code/.git/
+```
+
+### Working with a remote repository ###
+
+If you already have a remote repository to work with, you can do so by cloning the repository from GitHub onto your local machine.
+In order to start working locally on an existing remote repository, we need clone it with the command:
+
+```bash
+git clone <repository path>
+```
+You can either clone it via HTTPS or SSH, according to your preferred authentication method as we have shown above.
```

Comment:
Would add that HTTPS will soon be impossible if authentication is required (repo not public), and SSH already required if SSO is enforced on your github organisation.